http/vibe/http/auth/digest_auth.d

```diff
@@ -179,3 +180,132 @@ string createDigestPassword(string realm, string user, string password)
 }
 
 alias DigestHashCallback = string delegate(string realm, string user);
+
+/// Structure which describes requirements of the digest authentication - see https://tools.ietf.org/html/rfc2617
+struct DigestAuthParams {
+	enum Qop { none = 0, auth = 1, auth_int = 2 }
+	enum Algorithm { none = 0, md5 = 1, md5_sess = 2 }
+
+	string realm, domain, nonce, opaque;
+	Algorithm algorithm = Algorithm.md5;
+	bool stale;
+	Qop qop;
+
+	/// Parses WWW-Authenticate header value with the digest parameters
+	this(string auth) {
```

Comment:
A static method `static DigestAuthParams parse(in char[] auth)` would arguably be better here to make it clearer that actual parsing work is going on, and it's not just a field that is initialized.